devicetypes/smartthings/zigbee-thermostat.src/zigbee-thermostat.groovy

```diff
@@ -0,0 +1,337 @@
+	}
+}
+
+def parse(String description) {
+	def map = zigbee.getEvent(description)
+	def result
+	if(!map) {
+		result = parseAttrMessage(description)
+	} else {
+		log.warn "Some event is not parsed: ${map}"
+	}
+	log.debug "Description ${description} parsed to ${result}"
+	return result
+}
+
+private parseAttrMessage(description) {
+	def descMap = zigbee.parseDescriptionAsMap(description)
+	log.debug "Desc Map: $descMap"
+	def result = []
+	List attrData = [[cluster: descMap.clusterInt, attribute: descMap.attrInt, value: descMap.value]]
+	descMap.additionalAttrs.each {
+		attrData << [cluster: descMap.clusterInt, attribute: it.attrInt, value: it.value]
+	}
+	attrData.each {
+		def map = [:]
+		if (it.cluster == THERMOSTAT_CLUSTER && it.attribute == LOCAL_TEMPERATURE) {
+			log.debug "TEMP"
+			map.name = "temperature"
+			map.value = getTemperature(it.value)
+			map.unit = temperatureScale
+		} else if (it.cluster == THERMOSTAT_CLUSTER && it.attribute == COOLING_SETPOINT) {
+			log.debug "COOLING SETPOINT"
+			map.name = "coolingSetpoint"
+			map.value = getTemperature(it.value)
+			map.unit = temperatureScale
+		} else if (it.cluster == THERMOSTAT_CLUSTER && it.attribute == HEATING_SETPOINT) {
+			log.debug "HEATING SETPOINT"
+			map.name = "heatingSetpoint"
+			map.value = getTemperature(it.value)
+			map.unit = temperatureScale
+		} else if (it.cluster == THERMOSTAT_CLUSTER && (it.attribute == THERMOSTAT_MODE || it.attribute == THERMOSTAT_RUNNING_MODE)) {
+			log.debug "MODE"
+			map.name = "thermostatMode"
+			map.value = THERMOSTAT_MODE_MAP[it.value]
+		} else if (it.cluster == THERMOSTAT_CLUSTER && it.attribute == THERMOSTAT_RUNNING_STATE) {
+			log.debug "RUNNING STATE"
+			def binValue = extendString(bin(hexToInt(it.value)), 16, '0').reverse()
+			map.name = "thermostatOperatingState"
+			if(binValue[0] == "1") {
+				map.value = "heating"
+			} else if(binValue[1] == "1") {
+				map.value = "cooling"
+			} else {
+				map.value = binValue[2] == "1" ? "fan only" : "idle"
+			}
+		} else if (it.cluster == FAN_CONTROL_CLUSTER && it.attribute == FAN_MODE) {
+			log.debug "FAN MODE"
+			map.name = "thermostatFanMode"
+			map.value = FAN_MODE_MAP[it.value]
+		}
+		if(map) {
+			result << createEvent(map)
+		}
+	}
+	return result
+}
+
+def installed() {
+	refresh()
+}
+
+def refresh() {
+	return zigbee.readAttribute(THERMOSTAT_CLUSTER, LOCAL_TEMPERATURE) +
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, COOLING_SETPOINT) +
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, HEATING_SETPOINT) +
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, THERMOSTAT_MODE) +
+			zigbee.readAttribute(THERMOSTAT_CLUSTER, THERMOSTAT_RUNNING_STATE) +
+			zigbee.readAttribute(FAN_CONTROL_CLUSTER, FAN_MODE)
+}
+
+def ping() {
+	refresh()
+}
+
+def configure() {
+	def binding = zigbee.addBinding(THERMOSTAT_CLUSTER) + zigbee.addBinding(FAN_CONTROL_CLUSTER)
+	def startValues = zigbee.writeAttribute(THERMOSTAT_CLUSTER, HEATING_SETPOINT, DataType.INT16, 0x0A28) +
```

Comment:
Is there any reason these values were chosen?  The heating set point seems a little high.

Reply:
They're taken from ZCL reference but... It appears I've accidentally switched heating and cooling setpoints default values...

Reply:
Ah, I see now, they are the default values for those attributes.  If they are actually the defaults I wouldn't think we would have to set them on join, but I don't have any problem with it.  Swap them so they are the correct defaults and this PR looks good to me.